File: benchmark/results/python-multilang/run-004/b-baseline/output/metaengine_demo/catalog/services/product_service.py
"""ProductService for the catalog domain."""
from typing import Optional

from metaengine_demo.catalog.aggregates.product import Product


class ProductService:
    """Service exposing product operations."""

    def create(self, input: Optional[Product]) -> Product:
        """Create a new Product."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Product]:
        """Find a Product by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Product]:
        """List Products up to limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete a Product by id."""
        raise NotImplementedError("not implemented")

File: benchmark/results/ts-multilang/run-005/b-baseline/output/src/domain/catalog/services/ProductService.ts
import { Product } from "../aggregates/Product";

/** Application service for managing Product entities. */
export class ProductService {
  create(input: Partial<Product>): Product {
    void input;
    throw new Error("not implemented");
  }

  findById(id: string): Product | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): Product[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: string): void {
    void id;
    throw new Error("not implemented");
  }
}
